numpy/core/function_base.py

```diff
@@ -160,6 +160,9 @@ def linspace(start, stop, num=50, endpoint=True, retstep=False, dtype=None,
 
     if axis != 0:
         y = _nx.moveaxis(y, 0, axis)
+    
+    if _nx.issubdtype(dtype, np.integer):
+        y[y<0] = _nx.floor(y[y<0])
```

Comment:
Likely faster as just:
```suggestion
        _nx.floor(y, out=y)
```

Reply:
Thank you!